spec/unit/repositories_inspector_spec.rb

```diff
@@ -232,4 +226,124 @@
       expect(names).to eq(names.sort)
     end
   end
+
+  describe "yum repositories" do
+    let(:yum_conf) {<<-EOF
+[main]
+cachedir=/var/cache/yum/$basearch/$releasever
+keepcache=0
+debuglevel=2
+logfile=/var/log/yum.log
+exactarch=1
+obsoletes=1
+gpgcheck=1
+plugins=1
+installonly_limit=3
+
+[rhel-source]
+name=Red Hat Enterprise Linux $releasever - $basearch - Source
+baseurl=ftp://ftp.redhat.com/pub/redhat/linux/enterprise/$releasever/en/os/SRPMS/
+enabled=0
+gpgcheck=1
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release
+EOF
+    }
+
+    let(:find_yum_repos_output) {
+"/etc/yum.repos.d/download.opensuse.org_repositories_Virtualization_Appliances_RedHat_RHEL-6_.repo\n
+/etc/yum.repos.d/rhel-source.repo"
+    }
+
+    let(:yum_repo1) {<<-EOF
+
+[download.opensuse.org_repositories_Virtualization_Appliances_RedHat_RHEL-6_]
+name=added from: http://download.opensuse.org/repositories/Virtualization:/Appliances/RedHat_RHEL-6/
+baseurl=http://download.opensuse.org/repositories/Virtualization:/Appliances/RedHat_RHEL-6/
+enabled=1
+
+EOF
+    }
+
+    let(:yum_repo2) {<<-EOF
+
+[rhel-source-beta]
+name=Red Hat Enterprise Linux $releasever Beta - $basearch - Source
+baseurl=ftp://ftp.redhat.com/pub/redhat/linux/beta/$releasever/en/os/SRPMS/
+enabled=0
+gpgcheck=0
+gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-beta,file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release
+
+EOF
+    }
+
+    let(:expected_yum_repo_list) {
+      RepositoriesScope.new([
+        Repository.new(
+          alias: "download.opensuse.org_repositories_Virtualization_Appliances_RedHat_RHEL-6_",
+          name: "added from: http://download.opensuse.org/repositories/" \
+            "Virtualization:/Appliances/RedHat_RHEL-6/",
+          type: "rpm-md",
+          enabled: true,
+          gpgcheck: true,
+          url: "http://download.opensuse.org/repositories/Virtualization:/Appliances/RedHat_RHEL-6/"
+        ),
+        Repository.new(
+          alias: "rhel-source",
+          name: "Red Hat Enterprise Linux $releasever - $basearch - Source",
+          type: "rpm-md",
+          enabled: false,
+          gpgcheck: true,
+          url: "ftp://ftp.redhat.com/pub/redhat/linux/enterprise/$releasever/en/os/SRPMS/"
```

Comment:
I know but this seems to be derived from a redhat-release package which might be different on Fedora.